modules/components/style.js

```diff
@@ -24,18 +27,58 @@ var Style = React.createClass({
     }
   },
 
+  _buildStyles: function (stylesArr) {
+    var styles = reduce(stylesArr, function (accumulator, item) {
+      var selector = Object.keys(item)[0];
+      var rules = item[selector];
+
+      if (selector === "mediaQueries") {
+        accumulator += this._buildMediaQueryString(rules);
+      } else {
+        var completeSelector = (this.props.scopeSelector ?
+          this.props.scopeSelector + " " :
+          "") +
+          selector;
+        accumulator += buildCssString(completeSelector, rules);
+      }
+
+      return accumulator;
+    }, "", this);
+
+    return styles;
+  },
+
+  _buildMediaQueryString: function (mediaQueryObj) {
+    var mediaQueries = "";
+
+    Object.keys(mediaQueryObj).forEach(function (query) {
+      var completeQuery = this.contextMediaQueries[query] ?
+        this.contextMediaQueries[query] :
+        query;
+      mediaQueries += "@media " + completeQuery + "{" +
+        this._buildStyles(mediaQueryObj[query]) +
+      "}";
+    }.bind(this));
+
+    return mediaQueries;
+  },
+
+  _prepareContextMediaQueries: function () {
+    this.contextMediaQueries = {};
+    if (this.context && this.context.mediaQueries) {
+      Object.keys(this.context.mediaQueries).forEach(function (query) {
+        this.contextMediaQueries[query] = this.context.mediaQueries[query].media;
+      }.bind(this));
+    }
+  },
+
   render: function () {
     if (!this.props.rules) {
       return null;
     }
 
-    var styles = reduce(this.props.rules, function (s, item) {
-      var selector = Object.keys(item)[0];
-      var rules = item[selector];
-      var completeSelector = this.props.scopeSelector + " " + selector;
-
-      return s += buildCssString(completeSelector, rules);
-    }, "", this);
+    this._prepareContextMediaQueries();
```

Comment:
Would prefer to avoid caching additional data on `this`. Let's go for a more functional approach.